### ISSUE TITLE
Generate coverage report for pytest tests

### ISSUE THREAD

**Issue by erkist** [cdevents/sdk-python#3]
Add a Make target for generating test coverage reports, using [pytest-cov](https://pytest-cov.readthedocs.io/en/latest/readme.html) or similar

**Comment by erdepe-dad** [comment]
please assign me this issue, i'll working on it @erkist

**Comment by erkist** [comment]
Done, thank you @ahmadsyafrudin! Looking forward to the PR!

**Comment by erdepe-dad** [comment]
HI @erkist , Please Elaborate more about the goals. 
What I thought Is just added command to makefile that generate code coverage for this project. is that what we want ?
and also what is the report type that we want?
do we need code coverage badge on readme.md also?
last but not least, do you already have https://app.codecov.io/ account? or I can skip coverage badge part if we don't want it. 
Thanks 🙏🏻

**Comment by erdepe-dad** [comment]
#9 Kindly check and reveiw my pull request. thanks

**Comment by erkist** [comment]
@ahmadsyafrudin I'm very happy with what you have put in place already, as it allows me to get the following:

```shell
$ coverage report
Name                            Stmts   Miss  Cover
---------------------------------------------------
cdevents\core\__init__.py           1      0   100%
cdevents\core\artifact.py          20      0   100%
cdevents\core\branch.py            20      0   100%
cdevents\core\build.py             24      0   100%
cdevents\core\env.py               25      0   100%
cdevents\core\event.py             16      3    81%
cdevents\core\event_sender.py      11     11     0%
cdevents\core\event_type.py        24      0   100%
cdevents\core\http_handlar.py      95     95     0%
cdevents\core\pipelinerun.py       27      0   100%
cdevents\core\repository.py        25      0   100%
cdevents\core\service.py           29      0   100%
cdevents\core\taskrun.py           21      0   100%
---------------------------------------------------
TOTAL                             338    109    68%
```
As of right now I don't see a need for badges or codecov integration, so I will close this issue with your fix, but I am curious as to the next steps you have mentioned so I will create follow-up issues to look into them.

**Comment by erkist** [comment]
Solved by @ahmadsyafrudin in PR #9
